### PR TITLE
6528 remove incidentintent create

### LINKED ISSUE #6528: Don't write Incident:Create command anymore
**Description**

Instead of writing Incident:Create commands, the processing of the record (on which the incident occurs) should write Incident:Created directly.

It might make sense to postpone this change until all processors that may raise an incident have already been migrated.

### PR DESCRIPTION
**Pull request by pihme**
## Description

Removes the intent `IncidentIntend.CREATE` and everything related to it

## Related issues

closes #6528 

## Definition of Done

Code changes:
* [X] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [X] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: engine/src/test/java/io/zeebe/engine/processing/incident/JobFailIncidentTest.java [hunk 1]
```diff
@@ -123,20 +123,12 @@ public void shouldCreateIncidentIfJobHasNoRetriesLeft() {
...
-            .getFirst();
-
     final Record<IncidentRecordValue> incidentEvent =
         RecordingExporter.incidentRecords()
             .withIntent(IncidentIntent.CREATED)
             .withProcessInstanceKey(processInstanceKey)
             .getFirst();
 
-    assertThat(incidentCommand.getSourceRecordPosition())
-        .isEqualTo(failedEvent.getSourceRecordPosition());
```
Comment (korthout) [inline engine/src/test/java/io/zeebe/engine/processing/incident/JobFailIncidentTest.java:139]: I think we can verify the source record position of the incidentEvent instead. (please also note the other test case)

**Review by korthout — APPROVED** [review]
@pihme sorry for the late review. 

LGTM 🦩

**Comment by pihme** [comment]
bors retry

**Comment by pihme** [comment]
bors retry

**Pull request merged**